src/vnsw/agent/test/test_init.cc

```diff
@@ -17,26 +17,26 @@
 
 namespace opt = boost::program_options;
 
-pthread_t asio_thread;
+boost::thread asio_thread;
 
-void *asio_poll(void *arg){
+void *asio_poll() {
     Agent::GetInstance()->event_manager()->Run();
     return NULL;
 }
 
 void AsioRun() {
-    pthread_attr_t attr;
     int ret;
-
-    pthread_attr_init(&attr);
-    if ((ret = pthread_create(&asio_thread, &attr, asio_poll, NULL)) != 0) {
+    try {
+        asio_thread = boost::thread(asio_poll);
```

Comment:
are you sure it always works correctly? I mean they sometimes compile boost without exceptions, what would happen then?

Reply:
User provided throw_exception is going to be called. It actually means termination of an application.